pkg/volume/util/subpath/subpath_windows.go

```diff
@@ -44,31 +44,74 @@ func NewNSEnter(mounter mount.Interface, ne *nsenter.Nsenter, rootDir string) In
 	return nil
 }
 
-// evalPath returns the path name after the evaluation of any symbolic links.
-// If the path after evaluation starts with Volume or \??\Volume, it means that it was a symlink from
-// volume (represented by volumeID) to the given path. In this case, the given path is returned.
-func evalPath(path string) (linkedPath string, err error) {
-	cmd := fmt.Sprintf("Get-Item -Path %s | Select-Object -ExpandProperty Target", path)
+// evalSymlink returns the path name after the evaluation of any symbolic links.
+// If the path after evaluation is a device path or network connection, the original path is returned
+func evalSymlink(path string) (string, error) {
+	return evalWindowsSymlink(mount.NormalizeWindowsPath(path))
+}
+
+func evalWindowsSymlink(path string) (string, error) {
+	if isDeviceOrUncPath(path) || isDriveLetterorEmptyPath(path) {
+		klog.V(4).Infof("Path '%s' is not a symlink, return its original form.", path)
+		return path, nil
+	}
+	upperpath := path
+	base := ""
+	for {
```

Comment:
since it's a for loop, it's safer to set a max depth size, suggested max depth size: `256`

Reply:
good point. found information about directory length
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#maximum-path-length-limitation

So default maximum length is 260, but it could be extended with some requirements. The new limit is 32,767 characters. We could set it very conservatively to 32767, or some reasonable value in between?